awx/api/serializers.py

```diff
@@ -1412,6 +1412,18 @@ def get_related(self, obj):
             res['credential'] = self.reverse('api:credential_detail', kwargs={'pk': obj.credential.pk})
         return res
 
+    def validate(self, attrs):
+        # if ee already has organization, cannot change organization
+        # or set it to null
+        if self.instance and self.instance.organization_id is not None:
+            if 'organization' in attrs:
+                org = attrs.get('organization', None)
+                if not org:
+                    raise serializers.ValidationError({"organization": _("Cannot remove the execution environment from the organization")})
```

Comment:
Setting it to NULL should be OK. In this case you are moving from more restrictive to less restrictive.

Anyone can use EE w/ org Null.

Reply:
I concur.

Reply:
we currently allow going from null to non-null. So in essence a user could change from org-3 to org-2 by first setting to null, then setting to org-2.

Reply:
thanks, changed it to prevent going from null to an org. But can go from org to null